be/src/exec/cache_select_scanner.cpp

```diff
@@ -29,7 +29,12 @@ Status CacheSelectScanner::do_init(RuntimeState* runtime_state, const HdfsScanne
 Status CacheSelectScanner::do_open(RuntimeState* runtime_state) {
     // Cache select don't need to decompress data
     _compression_type = CompressionTypePB::NO_COMPRESSION;
-    RETURN_IF_ERROR(open_random_access_file());
+    // cache select only support text, parquet and orc format
+    if (_scanner_params.scan_range->file_format == THdfsFileFormat::TEXT ||
+        _scanner_params.scan_range->file_format == THdfsFileFormat::PARQUET ||
+        _scanner_params.scan_range->file_format == THdfsFileFormat::ORC) {
+        RETURN_IF_ERROR(open_random_access_file());
+    }
     return Status::OK();
 }
```

Comment:
For an unexpected format file, in fact we do not open it but return OK, it is a little tricky and easily misunderstood. 
We check the file format in `do_get_next` and return EOF directly. However, for other newly added features, relying on the return value of open function may result in a crash issue because using a null pointer `_file`.
So, why not check the file format in the upper layer and skip it for cache select scene directly?

Reply:
changed, PTAL

Reply:
> changed, PTAL

OK, please update the PR description about `What I'm doing` together.

Reply:
done